src/client-proxy.ts

```diff
@@ -0,0 +1,65 @@
+import { ClientProxy } from "@nestjs/microservices";
+import axios from "axios";
+import { resolve } from "dns";
+
+export class JSONRPCClient extends ClientProxy {
+  constructor(private readonly url: string) {
+    super();
+  }
+  private counter: number = 0;
+  private jsonrpc: string = "2.0";
+
+  connect(): Promise<any> {
+    throw new Error('The "connect()" method is not supported in JSONRPC mode.');
+  }
+  close() {
+    return Promise.resolve();
+  }
+  /**
+   * Method is unsupported for JSONRPC
+   */
+  protected publish(packet: any, callback: (packet: any) => any): any {
+    throw new Error("Method is not supported in JSONRPC mode.");
+  }
+  /**
+   * Method is unsupported for JSONRPC
+   */
+  protected async dispatchEvent(packet: any): Promise<any> {
+    throw new Error("Method is not supported in JSONRPC mode.");
+  }
+  getService<SvcInterface>(namespace: string): ServiceClient<SvcInterface> {
+    let url = this.url;
+    let id = this.counter;
+    let jsonrpc = this.jsonrpc;
+    return new Proxy(
+      {},
+      {
+        get(obj, prop) {
+          return function(params: any) {
+            return axios
+              .post(url, {
+                method: namespace + "." + prop.toString(),
+                params,
+                jsonrpc: "2.0",
+                id: ++id
```

Comment:
This will only increment the counter locally. `this.counter` will not be affected. We need to increment the counter itself